tests/groupRole.test.js

```diff
@@ -0,0 +1,1349 @@
+        assert.lengthOf(result.rows, 1);
+        assert.notEqual(result.rows[0].id, undefined);
+        locals.assessment_id = result.rows[0].id;
+        locals.assessmentUrl = locals.courseInstanceUrl + '/assessment/' + locals.assessment_id;
+        callback(null);
+      });
+    });
+    it('should contain a group-based homework assessment without roles', function (callback) {
+      sqldb.query(sql.select_group_work_assessment_without_roles, [], function (err, result) {
+        if (ERR(err, callback)) return;
+        assert.lengthOf(result.rows, 1);
+        assert.notEqual(result.rows[0].id, undefined);
+        assert.equal(result.rows[0].has_roles, false);
+        locals.assessment_id_without_roles = result.rows[0].id;
+        locals.assessmentUrlWithoutRoles =
+          locals.courseInstanceUrl + '/assessment/' + locals.assessment_id_without_roles;
+        callback(null);
+      });
+    });
+    it('should contain the 4 roles for the assessment', function (callback) {
+      const params = {
+        assessment_id: locals.assessment_id,
+      };
+      sqldb.query(sql.select_assessment_group_roles, params, function (err, result) {
+        if (ERR(err, callback)) return;
+        assert.lengthOf(result.rows, 4);
+        locals.groupRoles = result.rows;
+
+        // Store roles by name for later tests
+        const manager = result.rows.find((row) => row.role_name === 'Manager');
+        assert.isDefined(manager);
+
+        const recorder = result.rows.find((row) => row.role_name === 'Recorder');
+        assert.isDefined(recorder);
+
+        const reflector = result.rows.find((row) => row.role_name === 'Reflector');
+        assert.isDefined(reflector);
+
+        const contributor = result.rows.find((row) => row.role_name === 'Contributor');
+        assert.isDefined(contributor);
+
+        locals.manager = manager;
+        locals.recorder = recorder;
+        locals.reflector = reflector;
+        locals.contributor = contributor;
+        callback(null);
+      });
+    });
+  });
+
+  describe('2. get 4 student user', function () {
+    it('should insert/get 4 users into/from the DB', function (callback) {
+      sqldb.query(sql.generate_and_enroll_4_users, [], function (err, result) {
+        if (ERR(err, callback)) return;
+        assert.lengthOf(result.rows, 4);
+        locals.studentUsers = result.rows;
+        assert.lengthOf(locals.studentUsers, 4);
+        callback(null);
+      });
+    });
+
+    it('should switch current user to the group creator', function () {
+      config.authUid = locals.studentUsers[0].uid;
+      config.authName = locals.studentUsers[0].name;
+      config.authUin = '00000001';
+      config.userId = locals.studentUsers[0].user_id;
+    });
+  });
+
+  describe('3. POST to assessment page to create group', function () {
+    it('should load assessment page successfully', function (callback) {
+      request(locals.assessmentUrl, function (error, response, body) {
```

Comment:
This should use `node-fetch` and promises instead of `request` + callbacks. See https://github.com/PrairieLearn/PrairieLearn/pull/7206/files for an example of a PR where I made similar changes; note that that was with `request-promise-native` instead of just `request`, but the general idea should be the same.